target/linux/x86/base-files/usr/sbin/grow_rootfspart

```diff
@@ -0,0 +1,327 @@
+#!/bin/sh
+# shellcheck disable=SC2039
+#
+# Grow partition 2 (rootfs_data) until the end of the disk
+#
+# 
+
+readbin2hex() {
+	local input=$1
+	local seek=$2
+	local length=$3
+	hexdump -ve '1/1 "%.2x"' -s "$seek" -n "$length" "$input"
+}
+
+hex2bin(){
+	printf "%s\n" "$1" | sed -r 's/(.{2})/\\\\x\1 /g' | xargs printf '%b' 
+}
+
+writehex2bin(){
+	local output=$1
+	local seek=$2
+	local hexbuf=$3
+	local length=$((${#hexbuf}/2))
+	hex2bin "$hexbuf" | dd bs=1 seek="$seek" count="$length" of="$output" conv=notrunc 2>/dev/null
+}
+writehex2buf(){
+	local buf=$1
+	local seek=$2
+	local modified=$3
+	printf "%s" "${buf:0:seek*2}"
+	printf "%s" "$modified"
+	printf "%s\n" "${buf:seek*2+${#modified}}"
+}
+
+crc32_hex(){
+	gzip -c | tail -c8 | hexdump -n4 -e '"%.8x" "\n"'
+}
+
+
+msg(){
+        printf "grow_rootfspart: %s\n" "$*" >&2
+
+}
+die() {
+	local err=$1; shift
+	msg "$@"
+	exit "$err"
+}
+
+# swap endianness
+bswap(){
+	local hexle=$1
+	while [ "$hexle" ]; do
+		local last=$((${#hexle}-2))
+		echo -n "${hexle:$last:2}"
+		hexle=${hexle:0:$last}
+	done
+	echo
+}
+
+dec2hex(){ printf "%.$(($2*2))x\n" "$1"; }
+
+sector2chs() {
+	local sector=$1
+	if [ "$sector" -ge 16514064 ]; then
+		echo "ffffff"
+		return
+	fi
+	local c=$((sector / (HPC*SPT)))
+	local h=$(((sector / SPT) % HPC))
+	local s=$(((sector % SPT)))
+	b1=$(dec2hex "$h" 1)
+	b2=$(dec2hex "$(((c & 0x300 >> 2) | (s & 0x3f) ))" 1)
+	b3=$(dec2hex "$((c & 0xff))" 1)
+	printf "%s\n" "$b1$b2$b3"
+}
+
+mbr_parse_part() {
+	local partentry_hex=$1
+	#mbr_status=${partentry_hex:MBR_PARTSTATUS*2:MBR_PARTSTATUS_SIZE*2}
+	#mbr_first_chs=${partentry_hex:MBR_PARTFIRST_CHS*2:MBR_PARTCHS_SIZE*2}
+	#mbr_type=${partentry_hex:MBR_PARTTYPE*2:MBR_PARTTYPE_SIZE*2}
+	#mbr_last_chs=${partentry_hex:MBR_PARTLAST_CHS*2:MBR_PARTCHS_SIZE*2}
```

Comment:
Do we need these comments?

Reply:
No. Most of them where used only during the debug phase. I can remove all commented code. I kept them intentionally if someone might be interested on copying them to a library (like /lib/upgrade/common.sh).

Should I remove them?

Reply:
The script does what it does. I don't see much benefit by leftover debug code, even more since it's not commented.

Reply:
I removed all of them